### ISSUE TITLE
craetion of  rpm package  for  RHEL6

### ISSUE THREAD

**Issue by jherreraBE** [firehol/netdata#659]
hi,

I am  trying  to  generate a rpm  from the  spec file, but I have a  issue  with  linux/netfilter/nfnetlink_acct.h

when  I run  `rpmbuild -ba netdata.spec  --target=noarch` I have  the  following error:

```
In file included from plugin_nfacct.c:17:
/usr/include/libnetfilter_acct/libnetfilter_acct.h:6:44: error: linux/netfilter/nfnetlink_acct.h: No such file or directory
plugin_nfacct.c: In function 'nfacct_main':
plugin_nfacct.c:129: error: 'NFNL_MSG_ACCT_GET' undeclared (first use in this function)
plugin_nfacct.c:129: error: (Each undeclared identifier is reported only once
plugin_nfacct.c:129: error: for each function it appears in.)
make[2]: *** [plugin_nfacct.o] Error 1
make[2]: *** Waiting for unfinished jobs....
mv -f .deps/main.Tpo .deps/main.Po
mv -f .deps/plugin_proc.Tpo .deps/plugin_proc.Po
mv -f .deps/apps_plugin.Tpo .deps/apps_plugin.Po
make[2]: Leaving directory `/home/rpmbuild/rpmbuild/BUILD/netdata-1.2.0/src'
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory `/home/rpmbuild/rpmbuild/BUILD/netdata-1.2.0'
make: *** [all] Error 2
error: Bad exit status from /var/tmp/rpm-tmp.ACW5wL (%build)


RPM build errors:
    Bad exit status from /var/tmp/rpm-tmp.ACW5wL (%build)

I think that  it is becuase  I am missing  the  following  header  file  :   linux/netfilter/nfnetlink_acct.h

but  I have  already  install following  packages:

root@s51lv0e:/home/7514756 # ll  *.rpm
-rw-r--r-- 1 7514756 coreunix 18804 Jul  6 09:30 libnetfilter_acct-1.0.2-1.el6.x86_64.rpm
-rw-r--r-- 1 7514756 coreunix 22380 Jul  6 09:27 libnetfilter_acct-devel-1.0.2-1.el6.x86_64.rpm
root@s51lv0e:/home/7514756 #
```

so   I don't understand  .. why it is missing  ? do  I forgot other  package  to  install  to  be able  generate a rpm for  RHEL6.

thanks in advance,

Jonay

**Comment by ktsaou** [comment]
You should install `libmnl-dev`.

However, nfacct plugin will not work (I have to move it to plugin since it requires root access). So you can disable it (I don't know how to do this on rpmbuild).

**Comment by jherreraBE** [comment]
Hi  Costa,

thanks  a lot for  your  feedback. it helps  me further.. but now  I have
 the following  error,  if  you have  any idea  ..

thanks agian for your  assist  !!

kind  regards,

Jonay

 /usr/bin/install -c -m 644 images/seo-performance-16.png
images/seo-performance-24.png images/seo-performance-32.png
images/seo-performance-48.png images/seo-performance-64.png
images/seo-performance-72.png images/seo-performance-114.png
images/seo-performance-128.png images/seo-performance-256.png
images/seo-performance-512.png images/seo-performance-multi-size.ico
images/seo-performance-multi-size.icns
'/home/rpmbuild/rpmbuild/BUILDROOT/netdata-1.2.0-1.el6.noarch/usr/share/netdata/web/images'
 /usr/bin/install -c -m 644 lib/dygraph-combined.js
lib/dygraph-smooth-plotter.js lib/jquery-1.12.0.min.js
lib/jquery.peity.min.js lib/jquery.sparkline.min.js lib/morris.min.js
lib/raphael-min.js lib/jquery.easypiechart.min.js
lib/jquery.nanoscroller.min.js lib/bootstrap.min.js lib/ElementQueries.js
lib/ResizeSensor.js lib/bootstrap-toggle.min.js lib/c3.min.js lib/d3.min.js
lib/gauge.min.js
'/home/rpmbuild/rpmbuild/BUILDROOT/netdata-1.2.0-1.el6.noarch/usr/share/netdata/web/lib'
 /usr/bin/install -c -m 644 old/datasource.html old/index.html old/index.js
old/netdata.js old/theme.css
'/home/rpmbuild/rpmbuild/BUILDROOT/netdata-1.2.0-1.el6.noarch/usr/share/netdata/web/old'
 /bin/mkdir -p
'/home/rpmbuild/rpmbuild/BUILDROOT/netdata-1.2.0-1.el6.noarch/usr/share/netdata/web'
 /usr/bin/install -c -m 644 demo.html demo2.html demosites.html
dashboard.html dashboard.js dashboard.css dashboard.slate.css favicon.ico
index.html netdata-swagger.yaml netdata-swagger.json robots.txt tv.html
version.txt
'/home/rpmbuild/rpmbuild/BUILDROOT/netdata-1.2.0-1.el6.noarch/usr/share/netdata/web'
make[2]: Leaving directory `/home/rpmbuild/rpmbuild/BUILD/netdata-1.2.0/web'
make[1]: Leaving directory `/home/rpmbuild/rpmbuild/BUILD/netdata-1.2.0/web'
Making install in contrib
make[1]: Entering directory
`/home/rpmbuild/rpmbuild/BUILD/netdata-1.2.0/contrib'
make[2]: Entering directory
`/home/rpmbuild/rpmbuild/BUILD/netdata-1.2.0/contrib'
make[2]: Nothing to be done for `install-exec-am'.
make[2]: Nothing to be done for `install-data-am'.
make[2]: Leaving directory
`/home/rpmbuild/rpmbuild/BUILD/netdata-1.2.0/contrib'
make[1]: Leaving directory
`/home/rpmbuild/rpmbuild/BUILD/netdata-1.2.0/contrib'
make[1]: Entering directory `/home/rpmbuild/rpmbuild/BUILD/netdata-1.2.0'
make[2]: Entering directory `/home/rpmbuild/rpmbuild/BUILD/netdata-1.2.0'
make[2]: Nothing to be done for `install-exec-am'.
make[2]: Nothing to be done for `install-data-am'.
make[2]: Leaving directory `/home/rpmbuild/rpmbuild/BUILD/netdata-1.2.0'
make[1]: Leaving directory `/home/rpmbuild/rpmbuild/BUILD/netdata-1.2.0'
+ install -m 644 -p system/netdata.conf
/home/rpmbuild/rpmbuild/BUILDROOT/netdata-1.2.0-1.el6.noarch/etc/netdata/apps_groups.conf
/home/rpmbuild/rpmbuild/BUILDROOT/netdata-1.2.0-1.el6.noarch/etc/netdata/charts.d.conf
*install: target
`/home/rpmbuild/rpmbuild/BUILDROOT/netdata-1.2.0-1.el6.noarch/etc/netdata/charts.d.conf'
is not a directory*
*error: Bad exit status from /var/tmp/rpm-tmp.xEOeyO (%install)*


RPM build errors:
    Bad exit status from /var/tmp/rpm-tmp.xEOeyO (%install)


rpmbuild@s51lv0e:~/rpmbuild/BUILDROOT/netdata-1.2.0-1.el6.noarch/etc/netdata
$ cat  /var/tmp/rpm-tmp.xEOeyO
#!/bin/sh

  RPM_SOURCE_DIR="/home/rpmbuild/rpmbuild/SOURCES"
  RPM_BUILD_DIR="/home/rpmbuild/rpmbuild/BUILD"
  RPM_OPT_FLAGS="-O2"
  RPM_ARCH="noarch"
  RPM_OS="linux"
  export RPM_SOURCE_DIR RPM_BUILD_DIR RPM_OPT_FLAGS RPM_ARCH RPM_OS
  RPM_DOC_DIR="/usr/share/doc"
  export RPM_DOC_DIR
  RPM_PACKAGE_NAME="netdata"
  RPM_PACKAGE_VERSION="1.2.0"
  RPM_PACKAGE_RELEASE="1.el6"
  export RPM_PACKAGE_NAME RPM_PACKAGE_VERSION RPM_PACKAGE_RELEASE
  LANG=C
  export LANG
  unset CDPATH DISPLAY ||:

RPM_BUILD_ROOT="/home/rpmbuild/rpmbuild/BUILDROOT/netdata-1.2.0-1.el6.noarch"
  export RPM_BUILD_ROOT


PKG_CONFIG_PATH="${PKG_CONFIG_PATH}:/usr/lib64/pkgconfig:/usr/share/pkgconfig"
  export PKG_CONFIG_PATH

  set -x
  umask 022
  cd "/home/rpmbuild/rpmbuild/BUILD"
    [ "$RPM_BUILD_ROOT" != "/" ] && rm -rf "${RPM_BUILD_ROOT}"
    mkdir -p `dirname "$RPM_BUILD_ROOT"`
    mkdir "$RPM_BUILD_ROOT"

cd 'netdata-1.2.0'
LANG=C
export LANG
unset DISPLAY

rm -rf $RPM_BUILD_ROOT
/usr/bin/make -j4 install \
DESTDIR=$RPM_BUILD_ROOT

install -m 644 -p system/netdata.conf $RPM_BUILD_ROOT/etc/netdata/*.conf

find $RPM_BUILD_ROOT -name .keep | xargs rm




   /usr/lib/rpm/find-debuginfo.sh --strict-build-id
 "/home/rpmbuild/rpmbuild/BUILD/netdata-1.2.0"

    /usr/lib/rpm/check-buildroot

    /usr/lib/rpm/redhat/brp-compress

    /usr/lib/rpm/redhat/brp-strip-static-archive /usr/bin/strip
    /usr/lib/rpm/redhat/brp-strip-comment-note /usr/bin/strip
/usr/bin/objdump
    /usr/lib/rpm/brp-python-bytecompile
    /usr/lib/rpm/redhat/brp-python-hardlink
    /usr/lib/rpm/redhat/brp-java-repack-jars




rpmbuild@s51lv0e:~/rpmbuild/SPECS $ cd
 ../BUILDROOT/netdata-1.2.0-1.el6.noarch/etc/netdata/
rpmbuild@s51lv0e:~/rpmbuild/BUILDROOT/netdata-1.2.0-1.el6.noarch/etc/netdata
$ ll
total 8
-rw-r--r-- 1 rpmbuild rpmbuild 3184 Jul  7 17:13 apps_groups.conf
-rw-r--r-- 1 rpmbuild rpmbuild 1332 Jul  7 17:13 charts.d.conf
rpmbuild@s51lv0e:~/rpmbuild/BUILDROOT/netdata-1.2.0-1.el6.noarch/etc/netdata
$ less  charts.d.conf
rpmbuild@s51lv0e:~/rpmbuild/BUILDROOT/netdata-1.2.0-1.el6.noarch/etc/netdata
$ cat charts.d.conf
# This is the configuration for charts.d.plugin

# Each of its collectors can read configuration eiher from this file
# or a NAME.conf file (where NAME is the collector name).
# The collector specific file has higher precedence.

# This file is a shell script too.

#
-----------------------------------------------------------------------------

# number of seconds to run without restart
# after this time, charts.d.plugin will exit
# netdata will restart it, but a small gap
# will appear in the charts.d.plugin charts.
#restart_timeout=$[3600 * 4]

# when making iterations, charts.d can loop more frequently
# to prevent plugins missing iterations.
# this is a percentage relative to update_every to align its
# iterations.
# The minimum is 10%, the maximum 100%.
# So, if update_every is 1 second and time_divisor is 50,
# charts.d will iterate every 500ms.
# Charts will be called to collect data only if the time
# passed since the last time the collected data is equal or
# above their update_every.
#time_divisor=50

#
-----------------------------------------------------------------------------

# the default enable/disable for all charts.d collectors
#enable_all_charts="yes"

# per charts.d collector enable/disable
#nut=yes
#squid=yes
#postfix=yes
#sensors=yes
#cpufreq=yes
#mysql=yes
#example=yes
#load_average=yes
rpmbuild@s51lv0e:~/rpmbuild/BUILDROOT/netdata-1.2.0-1.el6.noarch/etc/netdata
$


On 6 July 2016 at 11:39, Costa Tsaousis <notifications@github.com> wrote:

> You should install libmnl-dev.
>
> However, nfacct plugin will not work (I have to move it to plugin since it
> requires root access). So you can disable it (I don't know how to do this
> on rpmbuild).
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/firehol/netdata/issues/659#issuecomment-230725500>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ATYCkYyQs5PSA0R3TaMJB3EAVCSQYQnKks5qS3fSgaJpZM4JF5XB>
> .
>



-- 
Groetjes - Regards,

Jonay Herrera y  Steendam

**Comment by ktsaou** [comment]
@philwhineray can you check this? I am out of ideas...

**Comment by paulfantom** [comment]
> find $RPM_BUILD_ROOT -name .keep | xargs rm

This will fail if `find` returns nothing. Error message:
```
rm: missing operand
Try 'rm --help' for more information
```

**Comment by jherreraBE** [comment]
hi,

thanks  for  the  feedback,  I have  make the  modification, but still the
 same   issue:

ke[2]: Nothing to be done for `install-data-am'.
make[2]: Leaving directory `/home/rpmbuild/rpmbuild/BUILD/netdata-1.2.0'
make[1]: Leaving directory `/home/rpmbuild/rpmbuild/BUILD/netdata-1.2.0'
+ install -m 644 -p system/netdata.conf
/home/rpmbuild/rpmbuild/BUILDROOT/netdata-1.2.0-1.el6.noarch/etc/netdata/apps_groups.conf
/home/rpmbuild/rpmbuild/BUILDROOT/netdata-1.2.0-1.el6.noarch/etc/netdata/charts.d.conf
install: target
`/home/rpmbuild/rpmbuild/BUILDROOT/netdata-1.2.0-1.el6.noarch/etc/netdata/charts.d.conf'
is not a directory
error: Bad exit status from /var/tmp/rpm-tmp.LNT66q (%install)


RPM build errors:
    Bad exit status from /var/tmp/rpm-tmp.LNT66q (%install)
rpmbuild@s51lv0e:~/rpmbuild/SPECS $ cat  /var/tmp/rpm-tmp.LNT66q
#!/bin/sh

  RPM_SOURCE_DIR="/home/rpmbuild/rpmbuild/SOURCES"
  RPM_BUILD_DIR="/home/rpmbuild/rpmbuild/BUILD"
  RPM_OPT_FLAGS="-O2"
  RPM_ARCH="noarch"
  RPM_OS="linux"
  export RPM_SOURCE_DIR RPM_BUILD_DIR RPM_OPT_FLAGS RPM_ARCH RPM_OS
  RPM_DOC_DIR="/usr/share/doc"
  export RPM_DOC_DIR
  RPM_PACKAGE_NAME="netdata"
  RPM_PACKAGE_VERSION="1.2.0"
  RPM_PACKAGE_RELEASE="1.el6"
  export RPM_PACKAGE_NAME RPM_PACKAGE_VERSION RPM_PACKAGE_RELEASE
  LANG=C
  export LANG
  unset CDPATH DISPLAY ||:

RPM_BUILD_ROOT="/home/rpmbuild/rpmbuild/BUILDROOT/netdata-1.2.0-1.el6.noarch"
  export RPM_BUILD_ROOT


PKG_CONFIG_PATH="${PKG_CONFIG_PATH}:/usr/lib64/pkgconfig:/usr/share/pkgconfig"
  export PKG_CONFIG_PATH

  set -x
  umask 022
  cd "/home/rpmbuild/rpmbuild/BUILD"
    [ "$RPM_BUILD_ROOT" != "/" ] && rm -rf "${RPM_BUILD_ROOT}"
    mkdir -p `dirname "$RPM_BUILD_ROOT"`
    mkdir "$RPM_BUILD_ROOT"

cd 'netdata-1.2.0'
LANG=C
export LANG
unset DISPLAY

rm -rf $RPM_BUILD_ROOT
/usr/bin/make -j4 install \
DESTDIR=$RPM_BUILD_ROOT

install -m 644 -p system/netdata.conf $RPM_BUILD_ROOT/etc/netdata/*.conf

find $RPM_BUILD_ROOT -name .keep -exec rm {} \;




   /usr/lib/rpm/find-debuginfo.sh --strict-build-id
 "/home/rpmbuild/rpmbuild/BUILD/netdata-1.2.0"

    /usr/lib/rpm/check-buildroot

    /usr/lib/rpm/redhat/brp-compress

    /usr/lib/rpm/redhat/brp-strip-static-archive /usr/bin/strip
    /usr/lib/rpm/redhat/brp-strip-comment-note /usr/bin/strip
/usr/bin/objdump
    /usr/lib/rpm/brp-python-bytecompile
    /usr/lib/rpm/redhat/brp-python-hardlink
    /usr/lib/rpm/redhat/brp-java-repack-jars

rpmbuild@s51lv0e:~/rpmbuild/SPECS $

kind  regards,

Jonay

On 8 July 2016 at 16:23, Paweł Krupa <notifications@github.com> wrote:

> find $RPM_BUILD_ROOT -name .keep | xargs rm
>
> This will fail if find returns nothing. Error message:
>
> rm: missing operand
> Try 'rm --help' for more information
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/firehol/netdata/issues/659#issuecomment-231372919>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ATYCkXTecmgb60JpvcgPUoY9j6vLolemks5qTl13gaJpZM4JF5XB>
> .
>



-- 
Groetjes - Regards,

Jonay Herrera y  Steendam

**Comment by paulfantom** [comment]
In line 8 of `netdata.spec.in` there is undefined variable named `with_systemd` but later it is checked against variable `with systemd`. Notice no `_` in checks.
Lines 65 and 74 should be changed to reflect this (change every `with systemd` to `with_systemd`).

Maybe this will help.

**Comment by paulfantom** [comment]
@ktsaou can we move `netdata.spec.in` to some folder? It doesn't feel right to have it in main dir since we have debian in `contrib`.

**Comment by ktsaou** [comment]
I don't know. @philwhineray do you know?

**Comment by paulfantom** [comment]
Line 26 is also wrong. It is `%if %{with systemd}`, but I think it should be `%if %{with_systemd}`.

**Comment by jherreraBE** [comment]
hi  guys,

thanks  all for   your  feedback, but it seems that this  version  of SPEC-file  is not usable. and  I am   not  so strong  in this.  I will  wait for  the  next version  of netdata  with a  working  SPEC-file. 
Or  has someone  one that is  working  ....

thanks  again,

jonay

**Comment by glensc** [comment]
use `rpmbuild --without systemd --without nfacct` to disable bconds (those defined with %bcond_xxx)

the %{with foo} and %{?with_foo} are equivalent. they are correct. look %with macro definition!

```
$ grep -r %with /usr/lib/rpm/
/usr/lib/rpm/macros.rpmbuild:%with()            %{expand:%%{?with_%{1}:1}%%{!?with_%{1}:0}}
/usr/lib/rpm/macros.rpmbuild:%without() %{expand:%%{?with_%{1}:0}%%{!?with_%{1}:1}}
```

also, attach your build output as files (.txt extension) or at least quote with triple backticks. it's unreadable!!!

**Comment by glensc** [comment]
also can't see full commands that were ran. the output is truncated and in wrong place.

also ` --target=noarch` is also nonsense, this package is not noarch!

**Comment by ktsaou** [comment]
I tried to add markdown for code blocks, but email replies seem to ignore that.

**Comment by glensc** [comment]
@ktsaou i wrote github support about that problem some time ago, they have not responded. they usually respond promptly.

**Comment by ktsaou** [comment]
@glensc ok, thanks!

**Comment by collinspam** [comment]
Unsubscribe me NOW

**Comment by jasonwbarnett** [comment]
@ktsaou I think this should be closed.

**Comment by paulfantom** [comment]
Closing since it is resolved in #728
